src/Swift.Bindings/src/Demangler/Swift5Reducer.cs

```diff
@@ -83,7 +91,21 @@ IReduction ConvertProtocolWitnessTable (Node node, string? name)
         var impNamed = (NamedTypeSpec)implementingType.TypeSpec;
         var protoNamed = (NamedTypeSpec)protocolType.TypeSpec;
 
-        return new ProtocolWitnessTableReduction() { Symbol = mangledName, ImplementingType = impNamed, ProtocolType = protoNamed };
+        if (isDescriptor) {
+            grandchild = Convert (child.Children [2]);
+            if (grandchild is ProvenanceReduction prov) {
+                if (!prov.Provenance.IsTopLevel)
+                    return ReductionError (ExpectedButGot ("A top-level module name", prov.Provenance.ToString()));
+                return new ProtocolConformanceDescriptorReduction() { Symbol = mangledName, ImplementingType = impNamed, ProtocolType = protoNamed, Module = prov.Provenance.Module};
+            } else if (grandchild is ReductionError) {
+                return grandchild;
+            } else {
+                return ReductionError (ExpectedButGot ("ProvenanceReduction", grandchild.GetType().Name));
```

Comment:
```suggestion
            } else {
                return ReductionError (ExpectedButGot ("ProvenanceReduction", grandchild.GetType().Name));
```

Would it be possible to unify this into `else` branch since we returning propagating `ReductionError` anyways.

Reply:
You don't know what error has come back from the previous call. By collapsing it, you're losing all information about the failure. Best to pass it back unchanged.